include/RevInstHelpers.h

```diff
@@ -76,15 +76,16 @@ bool load(RevFeature *F, RevRegFile *R, RevMem *M, RevInst Inst) {
     static constexpr auto flags = sizeof(T) < sizeof(int32_t) ?
       REVMEM_FLAGS(std::is_signed_v<T> ? RevCPU::RevFlag::F_SEXT32 : RevCPU::RevFlag::F_ZEXT32) :
       REVMEM_FLAGS(0);
-    req.Set(R->GetX<uint64_t>(Inst.rs1) + Inst.ImmSignExt(12),
+    uint64_t rs1 = R->GetX<uint64_t>(Inst.rs1); // read once for tracer
+    req.Set(rs1 + Inst.ImmSignExt(12),
             Inst.rd, RevRegClass::RegGPR,
             F->GetHartToExec(),
             MemOp::MemOpREAD,
             true,
             R->GetMarkLoadComplete());
     R->LSQueueInsert({make_lsq_hash(Inst.rd, RevRegClass::RegGPR, F->GetHartToExec()), req});
     M->ReadVal(F->GetHartToExec(),
-               R->GetX<uint64_t>(Inst.rs1) + Inst.ImmSignExt(12),
+               rs1 + Inst.ImmSignExt(12),
```

Comment:
What's the reason for this change? @leekillough can you confirm if this is okay or should be something else

Reply:
Oh wait I'm dumb. This is the same thing broken up into two stages so the tracer can use the value (I think)